### PR TITLE
[Dev] Expose app context header

### PR DESCRIPTION
**Pull request by zhoonit**
## Commits to be reviewed in this PR

<details><summary>[Dev] Expose app context header</summary><br />

This patch exposes app context header to make custom layer deployable.
Please note that this is a temporary measure and we might want to use
diffrent abstraction for this.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

<details><summary>[Header] Remove nntrainer_log.h from app_context.h</summary><br />

This patch removes nntrainer_log.h from app_context.h and implement
additional safecheck

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

### CODE REVIEW

File: debian/nntrainer-dev.install [hunk 1]
```diff
@@ -13,6 +13,8 @@
 /usr/include/nntrainer/layer_context.h
 /usr/include/nntrainer/layer_devel.h
 /usr/include/nntrainer/layer_impl.h
+# custom layer kits
+/usr/include/nntrainer/app_context.h
```
Comment (kparichay) [inline debian/nntrainer-dev.install:17]: this also requires nntrainer_log.h, or you can remove the dependency.

Comment (zhoonit) [reply debian/nntrainer-dev.install:17]: Yeah, I'll double check by installing this commit and if it compiles in the internal repo with including app context header

Comment (zhoonit) [reply debian/nntrainer-dev.install:17]: Now verified that this is working. I removed nntrainer_log.h dependency from app_context.h

**Review by kparichay — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**